server/bin/pbench-state-manager.py

```diff
@@ -61,15 +61,23 @@ def main(options):
         if options.create:
             args["owner"] = options.create
 
-        if options.path:
-            if not options.md5:
-                args["resource_id"] = md5sum(options.path).md5_hash
-            if options.create and not options.name:
-                args["name"] = Dataset.stem(options.path)
-        elif options.name and options.create:
-            args["name"] = options.name
+            if not (options.path or options.name):
+                print(
+                    f"{_NAME_}: Either --path or --name must be specified with --create",
+                    file=sys.stderr,
+                )
+                return 2
+
         if options.md5:
             args["resource_id"] = options.md5
+        elif options.path:
+            args["resource_id"] = md5sum(options.path).md5_hash
+
+        if options.create:
+            if options.name:
+                args["name"] = options.name
+            else:
+                args["name"] = Dataset.stem(options.path)
```

Comment:
Do you want to combine the bodies of the `if` statements at lines 61 and 76, since they have the same conditional?  (That would then allow you to combine the `if` statement at line 64 with the `else` at line 79 to produce a simpler `elif` test.)